x-pack/filebeat/input/http_endpoint/input.go

```diff
@@ -109,11 +117,18 @@ type pool struct {
 // cancelled or the context of another end-point sharing the same address
 // has had its context cancelled. If an end-point is re-registered with
 // the same address and mux pattern, serve will return an error.
-func (p *pool) serve(ctx v2.Context, e *httpEndpoint, pub stateless.Publisher) error {
+func (p *pool) serve(ctx v2.Context, e *httpEndpoint, pub stateless.Publisher, metrics *inputMetrics) error {
 	log := ctx.Logger.With("address", e.addr)
 	pattern := e.config.URL
 
-	var err error
+	metrics.bindAddr.Set(e.addr)
```

Comment:
I think this needs to come from the server. For example if through config I asked it to bind to (`127.0.0.1:0`) then the input metric should show the actual ephemeral port that it landed on.